src/pages/testPage/index.tsx

```diff
@@ -8,18 +8,34 @@ import { StylizedPaper } from '../../components/common/StylizedPaper/StylizedPap
 import { Timer } from '../../components/Timer/Timer';
 import { timeDefault } from '../../Mocs/TimerMock';
 import TestQuestions from './TestQuestions';
-import { testQuestions } from '../../Mocs/QuizMock';
 import { RectangleProgressTabs } from '../../components/common/Tabs/RectangleProgressTabs/RectangleProgressTabs';
+import { wrapper } from '../../store/store';
+import { questionsApi } from '../../api/questionsApi';
+import { fetchQuestionsAC } from '../../store/reducers/questions-reducer';
+import { TestQuestionsType } from '../../Types/TestQuestionsType';
 
-const TestPage = () => {
+export default function TestPage({
+  questions,
+}: {
+  questions: TestQuestionsType[];
+}) {
   const [currentTime, setCurrentTime] = useState(timeDefault);
   const [isRunning, setIsRunning] = useState(false);
+  const [numberOfQuestion, setQuestion] = useState(0);
+  const [disabled, setDisabled] = useState(false);
 
   const nextQuestionHandler = useCallback(() => {
-    setIsRunning(true);
-    setCurrentTime(timeDefault);
-    alert(`${currentTime.minutes}:${currentTime.seconds}`);
-  }, [currentTime]);
+    // setIsRunning(true);
+    // setCurrentTime(timeDefault);
+    // alert(`${currentTime.minutes}:${currentTime.seconds}`);
+
+    if (numberOfQuestion === questions.length - 1) {
```

Comment:
you don't need if else.
if (numberOfQuestion === questions.length - 1) {
      alert('The end');
      setDisabled(true);
      return
 }
setQuestion(numberOfQuestion + 1);

Reply:
I corrected it